diff --git b/Android.bp a/Android.bp
index b2cef8e..721b163 100644
--- b/Android.bp
+++ a/Android.bp
@@ -29,6 +29,10 @@ rust_test_host {
 rust_test {
     name: "unicode-xid_device_test_src_lib",
     defaults: ["unicode-xid_defaults"],
+    // override the following properties to work with coverage tests
+    test_suites: ["device-tests"],
+    auto_gen_config: false,
+    test_config: "AndroidTest.xml",
 }
 
 rust_defaults {
@@ -53,4 +57,8 @@ rust_test_host {
 rust_test {
     name: "unicode-xid_device_test_tests_exhaustive_tests",
     defaults: ["unicode-xid_defaults_exhaustive_tests"],
+    // override the following properties to work with coverage tests
+    test_suites: ["device-tests"],
+    auto_gen_config: false,
+    test_config: "AndroidTest_exhaustive_tests.xml",
 }
